In [70]:
import pandas as pd

# Cargar los archivos CSV con barras dobles
df_netflix_titles = pd.read_csv("C:\\Users\\fatim\\Downloads\\netflix_titles.csv")
df_netflix_originals = pd.read_csv("C:\\Users\\fatim\\Downloads\\netflix_originals.csv")

# Configurar Pandas para mostrar todas las filas y columnas
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

# Mostrar los DataFrame completos a través de display(df_netflix_originals) nd display(df_netflix_titles)

print("#### Dataframe de las producciones originales de Netflix ####")
display(df_netflix_originals.head())
display(df_netflix_originals.info())

print("#### Dataframe de los títulos adquiridos por Netflix ####")
display(df_netflix_titles.head())
display(df_netflix_titles.info())






#### Dataframe de las producciones originales de Netflix ####


Unnamed: 0            Title                  Genre           Premiere  \
0           0  Enter the Anime            Documentary     August 5, 2019   
1           1      Dark Forces               Thriller    August 21, 2020   
2           2          The App  Science fiction/Drama  December 26, 2019   
3           3   The Open House        Horror thriller   January 19, 2018   
4           4      Kaali Khuhi                Mystery   October 30, 2020   

   Runtime  IMDB Score          Language  
0       58         2.5  English/Japanese  
1       81         2.6           Spanish  
2       79         2.6           Italian  
3       94         3.2           English  
4       90         3.4             Hindi

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 513 entries, 0 to 512
Data columns (total 7 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Unnamed: 0  513 non-null    int64  
 1   Title       513 non-null    object 
 2   Genre       513 non-null    object 
 3   Premiere    513 non-null    object 
 4   Runtime     513 non-null    int64  
 5   IMDB Score  513 non-null    float64
 6   Language    513 non-null    object 
dtypes: float64(1), int64(2), object(4)
memory usage: 28.2+ KB


None

#### Dataframe de los títulos adquiridos por Netflix ####


Unnamed: 0 show_id     type                  title         director  \
0           0      s1    Movie   Dick Johnson Is Dead  Kirsten Johnson   
1           1      s2  TV Show          Blood & Water              NaN   
2           2      s3  TV Show              Ganglands  Julien Leclercq   
3           3      s4  TV Show  Jailbirds New Orleans              NaN   
4           4      s5  TV Show           Kota Factory              NaN   

                                                cast        country  \
0                                                NaN  United States   
1  Ama Qamata, Khosi Ngema, Gail Mabalane, Thaban...   South Africa   
2  Sami Bouajila, Tracy Gotoas, Samuel Jouy, Nabi...            NaN   
3                                                NaN            NaN   
4  Mayur More, Jitendra Kumar, Ranjan Raj, Alam K...          India   

           date_added  release_year rating   duration  \
0  September 25, 2021          2020  PG-13     90 min   
1  September 24, 2021          2021  TV-MA  2 Seasons   
2  September 24, 2021          2021  TV-MA        NaN   
3  September 24, 2021          2021  TV-MA        NaN   
4  September 24, 2021          2021  TV-MA  2 Seasons   

                                           listed_in  \
0                                      Documentaries   
1    International TV Shows, TV Dramas, TV Mysteries   
2  Crime TV Shows, International TV Shows, TV Act...   
3                             Docuseries, Reality TV   
4  International TV Shows, Romantic TV Shows, TV ...   

                                         description  
0  As her father nears the end of his life, filmm...  
1  After crossing paths at a party, a Cape Town t...  
2  To protect his family from a powerful drug lor...  
3  Feuds, flirtations and toilet talk go down amo...  
4  In a city of coaching centers known to train I...

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8807 entries, 0 to 8806
Data columns (total 13 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Unnamed: 0    8807 non-null   int64 
 1   show_id       8807 non-null   object
 2   type          8807 non-null   object
 3   title         8807 non-null   object
 4   director      6173 non-null   object
 5   cast          7982 non-null   object
 6   country       7976 non-null   object
 7   date_added    8797 non-null   object
 8   release_year  8807 non-null   int64 
 9   rating        8803 non-null   object
 10  duration      3994 non-null   object
 11  listed_in     8807 non-null   object
 12  description   8807 non-null   object
dtypes: int64(2), object(11)
memory usage: 894.6+ KB


None

In [71]:
print("#### Valores no nulos por columna - Netflix titles ####")
non_null_summary = pd.DataFrame({
    'Non-Null Count': df_netflix_titles.notnull().sum(),
    'Null Count': df_netflix_titles.isnull().sum(),
    'Non-Null Percentage': df_netflix_titles.notnull().mean() * 100
})
print(non_null_summary)

print("#### Valores no nulos por columna - Netflix originals ####")
non_null_summary = pd.DataFrame({
    'Non-Null Count': df_netflix_originals.notnull().sum(),
    'Null Count': df_netflix_originals.isnull().sum(),
    'Non-Null Percentage': df_netflix_originals.notnull().mean() * 100
})
print(non_null_summary)

#### Valores no nulos por columna - Netflix titles ####
              Non-Null Count  Null Count  Non-Null Percentage
Unnamed: 0              8807           0           100.000000
show_id                 8807           0           100.000000
type                    8807           0           100.000000
title                   8807           0           100.000000
director                6173        2634            70.091972
cast                    7982         825            90.632451
country                 7976         831            90.564324
date_added              8797          10            99.886454
release_year            8807           0           100.000000
rating                  8803           4            99.954582
duration                3994        4813            45.350290
listed_in               8807           0           100.000000
description             8807           0           100.000000
#### Valores no nulos por columna - Netflix originals ####
            Non-N

# Sobre nulos en Netflix Titles
Prioridades para corrección de datos nulos:

director: Rellenar con "Unknown" o imputar basándose en títulos similares.
country: Rellenar con "Unknown" o asignar valores basados en patrones (por ejemplo, directores o reparto).
duration: ¿solo aplicables a películas y omitible en el análisis de series?

# Sobre nulos en Netflix Originals
513 registros y todas las columnas completas.

In [12]:
import pandas as pd  

## Normalización de los datos
print("#### Normalización de los datos ####")

df_netflix_originals = pd.read_csv(r'C:\Users\fatim\Hackio\Laboratorios\Pandaslab\laboratorio-Pandas-produccionesdeNetflix\laboratorio-Pandas_produccionesdeNetflix\laboratorio-Pandas_Netflixproducciones\netflix_originals.csv')
df_netflix_titles = pd.read_csv(r'C:\Users\fatim\Hackio\Laboratorios\Pandaslab\laboratorio-Pandas-produccionesdeNetflix\laboratorio-Pandas_produccionesdeNetflix\laboratorio-Pandas_Netflixproducciones\netflix_titles.csv')


# Añadir una nueva columna 'is_original' con el valor True para identificar todos los registros del csv original
df_netflix_originals['is_original'] = True

# Verificar los resultados
display(df_netflix_originals.head())

## Gestión de valores nulos
print("#### Gestión de los valores nulos ####")
df_netflix_titles = df_netflix_titles.assign(
    director=df_netflix_titles['director'].fillna('Unknown'),
    cast=df_netflix_titles['cast'].fillna('Unknown'),
    country=df_netflix_titles['country'].fillna('Unknown'),
    duration=df_netflix_titles['duration'].fillna('No data')
)

# Actualización en el directorio y verificación de los cambios

df_netflix_originals.to_csv(r'C:\Users\fatim\Hackio\Laboratorios\Pandaslab\laboratorio-Pandas-produccionesdeNetflix\laboratorio-Pandas_produccionesdeNetflix\laboratorio-Pandas_Netflixproducciones\netflix_originals_updated.csv', index=False)
df_reloaded = pd.read_csv(r'C:\Users\fatim\Hackio\Laboratorios\Pandaslab\laboratorio-Pandas-produccionesdeNetflix\laboratorio-Pandas_produccionesdeNetflix\laboratorio-Pandas_Netflixproducciones\netflix_originals_updated.csv')
print(df_reloaded.columns)

df_netflix_titles.to_csv(r'C:\Users\fatim\Hackio\Laboratorios\Pandaslab\laboratorio-Pandas-produccionesdeNetflix\laboratorio-Pandas_produccionesdeNetflix\laboratorio-Pandas_Netflixproducciones\netflix_titles_updated.csv', index=False)
df_reloaded = pd.read_csv(r'C:\Users\fatim\Hackio\Laboratorios\Pandaslab\laboratorio-Pandas-produccionesdeNetflix\laboratorio-Pandas_produccionesdeNetflix\laboratorio-Pandas_Netflixproducciones\netflix_titles_updated.csv')
print(df_reloaded.columns)
display(df_netflix_titles.sample(6))


# ¿Por qué 'index'? evitar incluir el índice como una columna adicional.



#### Normalización de los datos ####


Unnamed: 0            Title                  Genre           Premiere  \
0           0  Enter the Anime            Documentary     August 5, 2019   
1           1      Dark Forces               Thriller    August 21, 2020   
2           2          The App  Science fiction/Drama  December 26, 2019   
3           3   The Open House        Horror thriller   January 19, 2018   
4           4      Kaali Khuhi                Mystery   October 30, 2020   

   Runtime  IMDB Score          Language  is_original  
0       58         2.5  English/Japanese         True  
1       81         2.6           Spanish         True  
2       79         2.6           Italian         True  
3       94         3.2           English         True  
4       90         3.4             Hindi         True

#### Gestión de los valores nulos ####
Index(['Unnamed: 0', 'Title', 'Genre', 'Premiere', 'Runtime', 'IMDB Score',
       'Language', 'is_original'],
      dtype='object')
Index(['Unnamed: 0', 'show_id', 'type', 'title', 'director', 'cast', 'country',
       'date_added', 'release_year', 'rating', 'duration', 'listed_in',
       'description'],
      dtype='object')


Unnamed: 0 show_id     type                            title  \
1894        1894   s1895  TV Show                   Emily in Paris   
1568        1568   s1569    Movie                        Kalel, 15   
5281        5281   s5282    Movie      Joaquín Reyes: Una y no más   
1587        1587   s1588  TV Show                        Detention   
786          786    s787    Movie                Company of Heroes   
3003        3003   s3004    Movie  Fortune Feimster: Sweet & Salty   

                                  director  \
1894                               Unknown   
1568                              Jun Lana   
5281                 José Miguel Contreras   
1587  Su I-Hsuan, Chuang Shiang-an, Liu Yi   
786                       Don Michael Paul   
3003                         Krysia Plonka   

                                                   cast        country  \
1894  Lily Collins, Philippine Leroy-Beaulieu, Ashle...  United States   
1568  Elijah Canlas, Eddie Garcia, Jaclyn Jose, Gabb...    Philippines   
5281                                      Joaquín Reyes        Unknown   
1587  Li Lingwei, Han Ning, Huang Guanzhi, Jack Yao,...         Taiwan   
786   Tom Sizemore, Chad Michael Collins, Vinnie Jon...  United States   
3003                                   Fortune Feimster  United States   

             date_added  release_year rating duration  \
1894    October 2, 2020          2020  TV-MA  No data   
1568   December 9, 2020          2019  TV-MA  No data   
5281  September 8, 2017          2017  TV-MA   78 min   
1587   December 5, 2020          2020  TV-MA  No data   
786        June 2, 2021          2013      R  No data   
3003   January 21, 2020          2020  TV-MA   61 min   

                                             listed_in  \
1894         Romantic TV Shows, TV Comedies, TV Dramas   
1568  Dramas, Independent Movies, International Movies   
5281                                   Stand-Up Comedy   
1587      International TV Shows, TV Dramas, TV Horror   
786                         Action & Adventure, Dramas   
3003                                   Stand-Up Comedy   

                                            description  
1894  After landing her dream job in Paris, Chicago ...  
1568  Surrounded by tensions and secrets, a teenage ...  
5281  Comedian and celebrity impersonator Joaquín Re...  
1587  A tormented student uncovers unsettling secret...  
786   During the last major German offensive of Worl...  
3003  Actor, comedian and writer Fortune Feimster ta...

In [8]:

    # Estadísticas descriptivas de las producciones originales de Netflix
    print("Estadísticas descriptivas de las producciones originales de Netflix")
    display(df_netflix_originals.describe(include='all').transpose()) #todas las columnas
    
    display(df_netflix_originals.describe(include = "object").T) #solo categóricas
    
    # Estadísticas descriptivas by titles
    print("Estadísticas descriptivas by titles")
    display(df_netflix_titles.describe(include='all').transpose()) #todas las columnas
    
    display(df_netflix_titles.describe(include = "object").T) #solo categóricas
    
    print(f"El número de filas que tenemos es {df_netflix_originals.shape[0]}, y el número de columnas es {df_netflix_titles.shape[1]}")


Estadísticas descriptivas de las producciones originales de Netflix


NameError: name 'df_netflix_originals' is not defined

### Diversidad de los contenidos originales. 
Se registran 513 producciones originales de Netflix de 106 géneros diferentes, siendo los documentales el género más frecuente (132 producciones). Esto muestra una clara preferencia o estrategia hacia este tipo de contenido. 
Además, se han traducido a 37 idiomas diferentes, con una predominancia clara del inglés (>350 producciones). Esto podría estar alineado con la intención de llegar a mercados internacionales, mientras se mantiene un enfoque en audiencias angloparlantes. 

La puntuación promedio (IMDb) alcanzada se sitúa en 6,21, una cifra que refleja una calidad moderada en términos de recepción crítica. Un rango amplio de puntuaciones, desde 2.5 hasta 9.0, indica variabilidad en la calidad percibida de las producciones. 

Duración media de las producciones (Runtime) es de 94,67 minutos, lo que indica una inclinación hacia películas de formato estándar. El rango oscila entre los 4 minutos (probablemente cortometrajes o contenido promocional) hasta 209 minutos (posiblemente los documentales). 



In [74]:
# Mostrar columnas de ambos DataFrames
print("Columnas en df_netflix_titles:")
print(df_netflix_titles.columns)

print("\nColumnas en df_netflix_originals:")
print(df_netflix_originals.columns)


# Realizar la unión interna por 'Unnamed'
merged_data = pd.merge(
    df_netflix_titles, 
    df_netflix_originals, 
    on='Unnamed: 0', 
    how='inner')


display(merged_df)

Columnas en df_netflix_titles:
Index(['Unnamed: 0', 'show_id', 'type', 'title', 'director', 'cast', 'country',
       'date_added', 'release_year', 'rating', 'duration', 'listed_in',
       'description'],
      dtype='object')

Columnas en df_netflix_originals:
Index(['Unnamed: 0', 'Title', 'Genre', 'Premiere', 'Runtime', 'IMDB Score',
       'Language', 'is_original'],
      dtype='object')


Unnamed: 0 show_id     type  \
0             0      s1    Movie   
1             1      s2  TV Show   
2             2      s3  TV Show   
3             3      s4  TV Show   
4             4      s5  TV Show   
5             5      s6  TV Show   
6             6      s7    Movie   
7             7      s8    Movie   
8             8      s9  TV Show   
9             9     s10    Movie   
10           11     s12  TV Show   
11           12     s13    Movie   
12           13     s14    Movie   
13           14     s15  TV Show   
14           15     s16  TV Show   
15           16     s17    Movie   
16           17     s18  TV Show   
17           18     s19    Movie   
18           19     s20  TV Show   
19           21     s22  TV Show   
20           22     s23    Movie   
21           23     s24    Movie   
22           24     s25    Movie   
23           25     s26  TV Show   
24           26     s27    Movie   
25           27     s28    Movie   
26           28     s29    Movie   
27           29     s30    Movie   
28           30     s31    Movie   
29           31     s32  TV Show   
30           32     s33  TV Show   
31           34     s35  TV Show   
32           35     s36    Movie   
33           36     s37    Movie   
34           37     s38  TV Show   
35           38     s39    Movie   
36           39     s40  TV Show   
37           40     s41  TV Show   
38           41     s42    Movie   
39           42     s43    Movie   
40           43     s44    Movie   
41           44     s45    Movie   
42           45     s46    Movie   
43           46     s47    Movie   
44           47     s48  TV Show   
45           48     s49    Movie   
46           49     s50  TV Show   
47           50     s51  TV Show   
48           51     s52    Movie   
49           52     s53    Movie   
50           53     s54    Movie   
51           54     s55    Movie   
52           55     s56  TV Show   
53           56     s57    Movie   
54           57     s58    Movie   
55           58     s59    Movie   
56           59     s60    Movie   
57           60     s61    Movie   
58           61     s62    Movie   
59           62     s63    Movie   
60           63     s64    Movie   
61           64     s65    Movie   
62           66     s67  TV Show   
63           67     s68  TV Show   
64           69     s70  TV Show   
65           70     s71  TV Show   
66           71     s72    Movie   
67           72     s73  TV Show   
68           73     s74    Movie   
69           74     s75  TV Show   
70           75     s76    Movie   
71           76     s77  TV Show   
72           77     s78    Movie   
73           78     s79    Movie   
74           80     s81    Movie   
75           81     s82    Movie   
76           82     s83  TV Show   
77           83     s84  TV Show   
78           84     s85    Movie   
79           85     s86  TV Show   
80           86     s87    Movie   
81           87     s88  TV Show   
82           88     s89    Movie   
83           89     s90  TV Show   
84           90     s91    Movie   
85           91     s92    Movie   
86           92     s93  TV Show   
87           93     s94    Movie   
88           94     s95    Movie   
89           95     s96  TV Show   
90           96     s97    Movie   
91           99    s100  TV Show   
92          100    s101  TV Show   
93          101    s102    Movie   
94          102    s103  TV Show   
95          103    s104    Movie   
96          104    s105  TV Show   
97          105    s106    Movie   
98          106    s107  TV Show   
99          107    s108    Movie   
100         108    s109  TV Show   
101         109    s110  TV Show   
102         110    s111  TV Show   
103         111    s112  TV Show   
104         112    s113    Movie   
105         113    s114    Movie   
106         114    s115    Movie   
107         115    s116    Movie   
108         117    s118    Movie   
109         118    s119    Movie   
110      

In [76]:
merged_data.to_csv(r'C:\Users\fatim\Downloads\laboratorio-Pandas_Netflixproducciones\data.netflix.csv', index=False)
import os
print(os.getcwd())


C:\Users\fatim
